hw_diag/utilities/dbus_proxy/modem_manager.py

```diff
@@ -0,0 +1,77 @@
+import dbus
+
+from hw_diag.utilities.dbus_proxy.dbus_ids import DBusIds
+from hw_diag.utilities.dbus_proxy.dbus_object import DBusObject
+from hw_diag.utilities.dbus_proxy.modem import Modem
+from hm_pyhelper.logger import get_logger
+
+logging = get_logger(__name__)
+
+
+class ModemManager(DBusObject):
+
+    def __init__(self, *args, **kwargs):
+        super(ModemManager, self).__init__(DBusIds.DBUS_MM1_SERVICE,
+                                           DBusIds.DBUS_MM1_PATH,
+                                           DBusIds.DBUS_MM1_IF)
+        self._mm_objmanager_iface = dbus.Interface(self._object_proxy,
+                                                   DBusIds.DBUS_OBJECTMANAGER_IF)
+
+    def get_all_modems(self):
+        '''
+        @rtype: list of all managed Modem objects
+        '''
+        modem_paths = self._mm_objmanager_iface.GetManagedObjects()
+        modems = [Modem(modem_path) for modem_path in modem_paths]
+        return modems
+
+    def _do_properties_match(self, actual_properties, desired_properties):
+        '''
+        return true if all properties in desired_properties are present
+        and same in actual_properties
+        '''
+        for prop_key in desired_properties:
+            actual_value = actual_properties.get(prop_key).strip()
+            desired_value = desired_properties.get(prop_key)
+            logging.debug(f'matching value of {prop_key} - '
+                          f'actual : {actual_value} and desired: {desired_value}')
+            if actual_value not in desired_value:
+                logging.info(f'actual value: {actual_value} does not match '
+                             f'any in desired: {desired_value}')
+                return False
+        # all matched
+        return True
+
+    def find_modem_by_properties(self, desired_properties):
+        '''
+        find modem for which all supplied properties match
+        @param properties: dict of properties to match
+        @rtype: Modem object or None if properties don't match any modem
+        '''
+        modems = self.get_all_modems()
+
+        # iterate over all object paths
+        for modem in modems:
+            props = modem.get_properties()
+            if self._do_properties_match(props, desired_properties):
+                return modem
+
+
+if __name__ == '__main__':
```

Comment:
These main methods are confusing. Its not clear if they are intended for production.

Reply:
i can remove those. But they help with quickly running code in ide.

Reply:
I don't have a strong preference about removing them. But if you keep them, we need to add documentation explaining their purpose and make the code production level (remove comments, unnecessary prints, etc). If you think the code will be useful in the future, then keep it, otherwise I don't see the value.

Reply:
I will remove all of them when I write test cases for these.